### PR TITLE
Add open in zeplin to sidebar

### PR DESCRIPTION
**Pull request by fonurr**
Add this option for projects, styleguides, component sections and components

### CODE REVIEW

File: src/common/vscode/message/MessageBuilder.ts [hunk 1]
```diff
@@ -29,7 +30,8 @@ export default class MessageBuilder {
     /**
      * Return VS Code's built-in message function.
      */
-    private getShowMessageFunction(): (message: string, ...items: string[]) => Thenable<string | undefined> {
+    private getShowMessageFunction():
+        (message: string, options: vscode.MessageOptions, ...items: string[]) => Thenable<string | undefined> {
```
Comment (konukselim) [inline src/common/vscode/message/MessageBuilder.ts:34]: Why did you create a private getter method for a function? Does your way any better than this one?

```suggestion
    private showMessage(message: string, options: vscode.MessageOptions, ...items: string[]) : Thenable<string | undefined> {
```

File: package.json [hunk 2]
```diff
@@ -74,7 +74,12 @@
             "view/item/context": [
                 {
                     "command": "zeplin.sidebar.removeBarrel",
-                    "when": "view == zeplin.views.barrels && viewItem == barrel"
+                    "when": "view == zeplin.views.barrels && viewItem =~ /\\bbarrel\\b/"
```
[the remark this answers is not in the document]
Comment (fonurr) [reply package.json:77]: It checks if it contains "barrel" as a word. In this pr I introduced supporting multiple contextValues in the format of "barrel/zeplinLink"

File: src/common/domain/openInZeplin/util/zeplinUrls.ts [hunk 3]
```diff
@@ -73,9 +88,30 @@ function getComponentWebUrl(barrelId: string, barrelType: BarrelType, componentI
...
+        ? getComponentSectionWebUrl(barrelId, barrelType, componentSectionId)
+        : getComponentSectionAppUrl(barrelId, barrelType, componentSectionId);
+}
+
+function getComponentSectionWebUrl(barrelId: string, barrelType: BarrelType, componentSectionId: string) {
+    const styleguideLabel = barrelType === BarrelType.Project ? "/styleguide" : "";
+    return `${getBarrelWebUrl(barrelId, barrelType)}${styleguideLabel}/components?seid=${componentSectionId}`;
+}
+
+function getComponentSectionAppUrl(barrelId: string, barrelType: BarrelType, componentSectionId: string) {
```
[the remark this answers is not in the document]
Comment (fonurr) [reply src/common/domain/openInZeplin/util/zeplinUrls.ts:104]: Oh, you're right. This should changed in many places. So I will open a PR for fixing all of them. Ignore this for now

File: src/common/vscode/message/MessageBuilder.ts (hunk 1 above)
Comment (fonurr) [reply src/common/vscode/message/MessageBuilder.ts:34]: You're right, I'll fix this

**Review by konukselim — APPROVED** [review]

**Review by muhammetalay — APPROVED** [review]

**Pull request merged**